weighted_levenshtein/clev.pxd

```diff
@@ -3,13 +3,13 @@ from libc.float cimport DBL_MAX as DTYPE_MAX
 ctypedef double DTYPE_t
 
 cdef enum:
-    ALPHABET_SIZE = 128
+    ALPHABET_SIZE = 512
 
 
 cdef DTYPE_t c_damerau_levenshtein(
-	unsigned char* str_a,
+	int[:] str_a,
```

Comment:
Since we are changing the API signatures, I think this is a breaking change for users who call c_damerau_levenshtein directly from Cython/C/C++, which is one of the supported and [documented](https://weighted-levenshtein.readthedocs.io/en/master/#use-as-cython-library) use cases of this library (though I don't know how many people actually use this feature).

Instead I think we should leave the existing Cython APIs with `unsigned char*`, and create new Cython APIs with `int[:]`.
The Python APIs can remain the same.

Reply:
Are you suggesting leaving the original code as well? Wouldn't that make the code too bloated?

Reply:
I guess what we can do is a translation layer.

We now expose 2 sets of public APIs: one set using `int[:]`, and one set using `unsigned char*`.

However, you can internally call the `int[:]` version from the `unsigned char*` version. This allows you to support both APIs using only one implementation.

Reply:
Good idea, I'll try that!

Reply:
Since I haven't used Cython before, I couldn't convert `unsigned char*` to  `int[:]`.
Instead, I just copied over the legacy code.
Is that alright?

Reply:
What if we use `unsigned int*` (`unsigned` to be consistent with `unsigned char*`) instead of `int[:]`?

The problem with int[:] is that you can't create it without holding the GIL [[source](https://groups.google.com/forum/#!msg/cython-users/rEd-1Pe8jKw/Fk0ulF0D9zkJ)], which we would like to avoid because it hurts performance. But if you used `malloc`/`free` with `unsigned int*` you won't need to hold the GIL.

See [here](https://github.com/infoscout/weighted-levenshtein/blob/master/weighted_levenshtein/clev.pyx#L39) for how we use malloc/free right now.


You could do something like this:

```
cdef DTYPE_t levenshtein(unsigned char* str1, Py_ssize_t len1, unsigned char* str2, Py_ssize_t len2, ...) nogil:
    cdef unsigned int* int_arr1 = <unsigned int*> malloc(len1 * sizeof(unsigned int))
    copy_str_to_int_arr(str1, len1, int_arr1)
    cdef unsigned int* int_arr2 = <unsigned int*> malloc(len2 * sizeof(unsigned int))
    copy_str_to_int_arr(str2, len2, int_arr2)

    DTYPE_t result = levenshtein_unicode(int_arr1, len1, int_arr2, len2, ...)
    
    free(int_arr1)
    free(int_arr2)
    return result


cdef void copy_str_to_int_arr(unsigned char* str, Py_ssize_t len, unsigned int* int_arr) nogil:
    for i in range(len):
        int_arr[i] = str[i]


cdef DTYPE_t levenshtein_unicode(unsigned int* str1, Py_ssize_t len1, unsigned int* str2, Py_ssize_t len2, ...) nogil:
    ...
```

Reply:
Done! The code is smaller now.